### PR TITLE
data service log bug

### LINKED ISSUE #431: Data service doesn't log Cosmos connection error
log error message if data service fails to start when connecting to cosmos. it currently fails without any log messages.

### PR DESCRIPTION
**Pull request by bartr**
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- data service doesn't log an error when it can't connect to Cosmos

## Validation

- [x] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #431

### CODE REVIEW

**Comment by kforeverisback** [comment]
Tested locally for three cases:
1. Wrong Cosmos URL
2. Wrong Cosmos Key
3. Cosmos exists but SQL container wasn't created

All tests printed exception to stderr

File: src/Ngsa.DataService/Core/CommandLine.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public static RootCommand BuildRootCommand()
             root.AddOption(new Option<int>(new string[] { "--perf-cache" }, "Cache only when load exceeds value"));
             root.AddOption(new Option<string>(new string[] { "--secrets-volume" }, () => "secrets", "Secrets Volume Path"));
             root.AddOption(new Option<LogLevel>(new string[] { "-l", "--log-level" }, () => LogLevel.Warning, "Log Level"));
-            root.AddOption(new Option<bool>(new string[] { "-d", "--dry-run" }, "Validates configuration"));
+            root.AddOption(new Option<bool>(new string[] { "-d", "--dry-run" }, "Validrates configuration"));
```
Comment (aflinchb) [inline src/Ngsa.DataService/Core/CommandLine.cs:72]: Small typo snuck in :) "Validrates"

**Review by aflinchb — COMMENTED** [review]
It wasn't clear to me how this fixed the bug https://github.com/retaildevcrews/ngsa/issues/431 (Data service doesn't log Cosmos connection error) - was it due to the logic error when comparing log levels?

Small comment on typo too.

**Comment by bartr** [comment]
yes, when I converted the code, I used >= instead of <= ...

**Review by aflinchb — APPROVED** [review]

**Pull request merged**